src/core/config.c

```diff
@@ -115,30 +115,56 @@ static char *get_config_path(void) {
 	prefix[0] = config_home;
 	prefix[1] = SYSCONFDIR "/xdg";
 
-	const char *config[2];
-	config[0] = getenv("XDG_CURRENT_DESKTOP");
-	config[1] = "config";
+	int config_num;
+	char *xdg_current_desktop = getenv("XDG_CURRENT_DESKTOP");
+	if (xdg_current_desktop == NULL || xdg_current_desktop[0] == '\0') {
+		config_num = 1;
+		xdg_current_desktop = NULL;
+	} else {
+		config_num = 2;
+		for (size_t i = 0; xdg_current_desktop[i] != '\0'; i++) {
+			if (xdg_current_desktop[i] == ':')
+				config_num++;
+		}
+		xdg_current_desktop = strdup(xdg_current_desktop);
+	}
+
+	const char *config[config_num];
+	if (xdg_current_desktop != NULL) {
+		char *token = strtok(xdg_current_desktop, ":");
+		for (int i = 0; i < config_num - 1; i++) {
+			logprint(TRACE, "xdpw: token: %s (%p)", token, token);
+			config[i] = token;
+			token = strtok(NULL, ":");
```

Comment:
1. Is strtok nilpotent (safe to call arbitrary often)?
2. Why is strtok "skipping" delimiters?
    ```
    % XDG_CURRENT_DESKTOP="::sway:wlroots" ./build/xdg-desktop-portal-wlr -l TRACE  
    2021/04/23 01:12:16 [TRACE] - xdpw: token: sway (0x602000000010)
    2021/04/23 01:12:16 [TRACE] - xdpw: token: wlroots (0x602000000015)
    2021/04/23 01:12:16 [TRACE] - xdpw: token: (null) ((nil))
    2021/04/23 01:12:16 [TRACE] - xdpw: token: (null) ((nil))
    2021/04/23 01:12:16 [TRACE] - config: trying config file ~/.config/xdg-desktop-portal-wlr/sway
    2021/04/23 01:12:16 [TRACE] - config: trying config file ~/.config/xdg-desktop-portal-wlr/wlroots
    2021/04/23 01:12:16 [TRACE] - config: trying config file ~/.config/xdg-desktop-portal-wlr/config
    2021/04/23 01:12:16 [TRACE] - config: trying config file /usr/local/etc/xdg/xdg-desktop-portal-wlr/sway
    2021/04/23 01:12:16 [TRACE] - config: trying config file /usr/local/etc/xdg/xdg-desktop-portal-wlr/wlroots
    2021/04/23 01:12:16 [TRACE] - config: trying config file /usr/local/etc/xdg/xdg-desktop-portal-wlr/config
    ```

Reply:
1. should be according to the manpage
2. default behaviour according to the manpage